indexes/utxoindex/src/stores/store_manager.rs

```diff
@@ -1,98 +1,114 @@
-use parking_lot::RwLock;
 use std::sync::Arc;
 
 use consensus_core::{tx::ScriptPublicKeys, BlockHashSet};
 use database::prelude::{StoreError, DB};
 use kaspa_core::trace;
 
 use crate::{
-    errors::UtxoIndexError,
-    model::{UtxoChanges, UtxoSetByScriptPublicKey},
+    model::UtxoSetByScriptPublicKey,
     stores::{
         indexed_utxos::{DbUtxoSetByScriptPublicKeyStore, UtxoSetByScriptPublicKeyStore, UtxoSetByScriptPublicKeyStoreReader},
         supply::{CirculatingSupplyStore, CirculatingSupplyStoreReader, DbCirculatingSupplyStore},
         tips::{DbUtxoIndexTipsStore, UtxoIndexTipsStore, UtxoIndexTipsStoreReader},
     },
+    IDENT,
 };
 
 #[derive(Clone)]
-pub struct StoreManager {
-    utxoindex_tips_store: Arc<RwLock<DbUtxoIndexTipsStore>>,
-    circulating_suppy_store: Arc<RwLock<DbCirculatingSupplyStore>>,
-    utxos_by_script_public_key_store: Arc<RwLock<DbUtxoSetByScriptPublicKeyStore>>,
+pub struct Store {
+    utxoindex_tips_store: DbUtxoIndexTipsStore,
+    circulating_supply_store: DbCirculatingSupplyStore,
+    utxos_by_script_public_key_store: DbUtxoSetByScriptPublicKeyStore,
 }
 
-impl StoreManager {
+impl Store {
     pub fn new(db: Arc<DB>) -> Self {
         Self {
-            utxoindex_tips_store: Arc::new(RwLock::new(DbUtxoIndexTipsStore::new(db.clone()))),
-            circulating_suppy_store: Arc::new(RwLock::new(DbCirculatingSupplyStore::new(db.clone()))),
-            utxos_by_script_public_key_store: Arc::new(RwLock::new(DbUtxoSetByScriptPublicKeyStore::new(db, 0))),
+            utxoindex_tips_store: DbUtxoIndexTipsStore::new(db.clone()),
+            circulating_supply_store: DbCirculatingSupplyStore::new(db.clone()),
+            utxos_by_script_public_key_store: DbUtxoSetByScriptPublicKeyStore::new(db, 0),
         }
     }
 
     pub fn get_utxos_by_script_public_key(
         &self,
         script_public_keys: &ScriptPublicKeys,
     ) -> Result<UtxoSetByScriptPublicKey, StoreError> {
-        let reader = self.utxos_by_script_public_key_store.read();
-        reader.get_utxos_from_script_public_keys(script_public_keys)
+        self.utxos_by_script_public_key_store.get_utxos_from_script_public_keys(script_public_keys)
     }
 
-    pub fn get_all_utxos(&self) -> Result<UtxoSetByScriptPublicKey, StoreError> {
-        let reader = self.utxos_by_script_public_key_store.read();
-        reader.get_all_utxos()
-    }
+    pub fn update_utxo_state(
+        &mut self,
+        to_add: Option<&UtxoSetByScriptPublicKey>,
```

Comment:
The API will be clearer if you'll pass an empty hashmap instead of `None` (and then in `add_utxo_entries` you can exit early if it's empty)